### ISSUE TITLE
Hangs loading in rpy2 submodule (works in notebook)

### ISSUE THREAD

**Issue by NicWayand** [jupyterlab/jupyterlab#4817]
The package `rpy2` loads fine: 
```python 
import rpy2
```

but when I try to load the submodule robjects:
```python
import rpy2.robjects as robjects
```
the kernal restarts (hovering over the dot says "kernal starting")

This does not happen when using notebook, just jupyterlab. I am not sure if this belongs here or in a rpy2 bitbucket issue, but since it works in notebook I decided to start here.

jupyterlab  0.32.1

**Comment by jasongrout** [comment]
Does it work from a command line ipython console?

**Comment by jasongrout** [comment]
Just saw your note about the env. That does seem weird.

**Comment by NicWayand** [comment]
Yep, if I am on the rpy2 env, and start ipython via the console it imports fine.

Agreed! I am fine running two jupyter lab instances on each env for now. But it would be nice to know why it restarts. Is there a way to get more verbose logging?

**Comment by BoPeng** [comment]
I was a long time user of `rpy` and `rpy2` and even helped the maintenance of `rpy` but I realized that such cross-language modules bring more troubles than benefits to my work. More specifically, I had to go through a lot of hooks to convert objects, parameter names, dealing with special cases and weird crashes and memory problems, and it was a lot of work to distribute `rpy2` with my python module because it is version-dependent on R. Eventually I decided to bridge data, not scripts, and wrote Script of Scripts (https://vatlab.github.io/sos-docs/). :smile:

**Comment by aok1425** [comment]
This _may_ answer your question: https://github.com/jupyter/notebook/issues/1841#issuecomment-263383793

**Comment by rousso1** [comment]
I'm experiencing the same issue. [see here](https://bitbucket.org/rpy2/rpy2/issues/214/segmentation-fault-11-on-mac-os-x-mountain#comment-49180043)

**Comment by lgautier** [comment]
@integratebio : what do you mean by "I had used the path" ? By default rpy2 will look for the first R found in the PATH. What can happen is that the R in the PATH does not match the R shared library picked up when dynamically loading it. Setting LD_LIBRARY_PATH according can help. There is a utility in the package that can tell you what its value should be:

```bash
python -m rpy2.situation LD_LIBRARY_PATH
```

Also, do you happen to use `conda`? Most such issues I have heard about originated from conda setting up PATH, LD_LIBRARY_PATH, or compiling with `-rpath` in a way that either confused the user (or was self-inconsistent).

**Comment by lielieee** [comment]
In windows system, I use:

import os 
os.environ['R_HOME'] = 'D:\\xxx\\xxx\\Lib\\R'    #path to your R installation
# must be "\\"......

**Comment by imanzabet** [comment]
You need to set `R_HOME` in the environment variable.
Or simply use this code:
`import os
os.environ['R_HOME'] = 'C:/Program Files/R/R-4.3.3'
`
